Sources_v3/StreamChat/Workers/MessageUpdater.swift

```diff
@@ -61,7 +61,7 @@ class MessageUpdater<ExtraData: ExtraDataTypes>: Worker {
             }
             
             if messageDTO.existsOnlyLocally {
-                session.delete(message: messageDTO)
+                messageDTO.deletedAt = Date()
```

Comment:
Please explain what's happening here and why

Reply:
Yep, that's the tricky one.

We get the crash trying to map `MessageDTO -> ChatMessage` when we catch **delete** event by FRC because after the deletion the relationships become **nil** and we cannot parse `author: ChatUser`.

We faced with this issue working on DB flush when current user is changed. If you remember, setting delete rule to `NoAction` helps to get object with **non-nil** relations but I don't think having invalid object graph is good way to go.

In this particular case, soft delete works nicely: for local message the same behavior as for sent message is applied (the message stays in the UI but with `The message is deleted` content and `is only visible to you`)

Reply:
updates to change state to `deleted` in 2f912ffd906923629b40d5645b123478e4a23d4e